image/templates/helm/shared/templates/02-scanner-v4-00-serviceaccount.yaml

```diff
@@ -0,0 +1,16 @@
+{{- include "srox.init" . -}}
```

Comment:
Before we handed over the work to Draco team, we kept them separate for readability. Yes we now have many more files, but, to me at least, it's way easier to read than parsing through a bunch of conditionals in files, which we plan on getting rid of in a few releases anyway. If there is a clean and readable way to do this, though, then I'm all for it

Reply:
I understand both points here. And I think one of the underlying issues here is that there is no concrete plan yet for some kind of consolidation of installation methods + simplification of Helm charts and templating code. At least nothing that I am aware of. Yes, there are ideas floating around. And reducing templating logic has been raised several times. As far as I am concerned, I feel like there is not a common understanding yet about some kind of "templating complexity sweet spot".

Until some steps have been made towards this, I would suggest to go with a very naive approach: Say, if it only requires something like one or two (ideally) top-level conditionals to be able to share a manifest, then we should do it.

Reply:
tbh, I'd like to take this back. I was assuming that there is a straight-forward mechanism for saying "include this file here and overwrite its metadata name like so and so". But, there isn't. I mean, yes, one can move the content of something like `image/templates/helm/shared/templates/02-scanner-00-serviceaccount.yaml` to `image/templates/helm/shared/templates/generic-service-account.tpl`, enclose the content in `{{ define "generic-service-account" ... }} ... {{ end }}`, add some logic for passing in additional parameters for setting the name of the resource and finally use `{{ include "generic-service-account" (list "scanner" .) }}` or something in one YAML and `{{ include "generic-service-account" (list "scanner-v4" .) }}` in another YAML.

I have my doubts that this would be a good cost/benefit ratio, *especially* given that we would like to reduce Helm complexity.
WDYT, @johannes94?

Reply:
What I was suggesting goes more in the direction of sharing the K8s resource not the template.

E.g. instead of having a `scanner` Role assigned to a `scanner` ServiceAccount and a `scanner-v4` Role assigned to a `scanner-v4` ServiceAccount. We could have a single `scanner` role assigned to a single `scanner` ServiceAccount that is referenced by `scanner-deployment` and `scanner-v4-deployment`.